core/server/worker/src/main/java/alluxio/worker/page/PagedBlockStore.java

```diff
@@ -507,4 +509,29 @@ private int getDirIndexOfBlock(long blockId) {
         .getDir()
         .getDirIndex();
   }
+
+  private void removeBlockInternal(long sessionId, long blockId, long timeoutMs)
+      throws IOException {
+    Optional<BlockLock> optionalLock =
+        mLockManager.tryAcquireBlockLock(sessionId, blockId, BlockLockType.WRITE,
+        timeoutMs, TimeUnit.MILLISECONDS);
+    if (!optionalLock.isPresent()) {
+      throw new DeadlineExceededException("Cannot acquire lock to remove block");
+    }
+    try (BlockLock blockLock = optionalLock.get()) {
+      Set<PageId> pageIds;
+      try (LockResource metaLock = new LockResource(mPageMetaStore.getLock().readLock())) {
```

Comment:
You need a write lock to delete blocks and pages.

Reply:
@dbw9580 done